src/components/QueryManager.jsx

```diff
@@ -0,0 +1,182 @@
+/* eslint-disable no-template-curly-in-string */
+// React imports
+import React from "react";
+import PropTypes from "prop-types";
+
+const QueryManager = ({
+  search,
+  name,
+  unix,
+  country,
+  state,
+  city,
+  year27Selected,
+  year26Selected,
+  year25Selected,
+  year24Selected,
+  professorSelected,
+  staffSelected,
+  studentSelected,
+  building,
+  advancedFiltersSelected,
+  updateQuery,
+}) => {
+  // display all active filters
+  let str = "";
+  let prior = false;
+
+  // base query (search box)
+  if (search.trim() !== "") {
+    str += search;
+    prior = true;
+  }
+
+  // name
+  if (name.trim() !== "") {
+    str += prior ? " AND (" : "(";
+    str += `name: "${name}")`;
+    prior = true;
+  }
+
+  // unix
+  if (unix.trim() !== "") {
+    str += prior ? " AND (" : "(";
+    str += `unix: "${unix}")`;
+    prior = true;
+  }
+
+  // country
+  if (state !== "") country = "United States";
+  if (country.trim() !== "") {
+    str += prior ? " AND (" : "(";
+    str += `country: "${country}")`;
+    prior = true;
+  }
+
+  // state
+  if (state !== "") {
+    str += prior ? " AND (" : "(";
+    str += `state: "${state}")`;
+    prior = true;
+  }
+
+  // town
+  if (city.trim() !== "") {
+    str += prior ? " AND (" : "(";
+    str += `city: "${city}")`;
+    prior = true;
+  }
+
+  // year
+  let firstYear = 1;
+  if (!year27Selected) {
+    firstYear = 2;
+    if (!year26Selected) {
+      firstYear = 3;
+      if (!year25Selected) {
+        firstYear = 4;
+      }
+    }
+  }
+  if (year27Selected || year26Selected || year25Selected || year24Selected)
+    str += prior ? " AND (" : "(";
+  if (year27Selected) str += 'year: "27"';
+  if (year26Selected) str += firstYear === 2 ? 'year: "26"' : ' OR year: "26"';
+  if (year25Selected) str += firstYear === 3 ? 'year: "25"' : ' OR year: "25"';
+  if (year24Selected) str += firstYear === 4 ? 'year: "24"' : ' OR year: "24"';
+  if (year27Selected || year26Selected || year25Selected || year24Selected) {
+    str += ")";
+    prior = true;
+  }
```

Comment:
As in previous comment, turn the state into a mathematical term to be added to the base academic year.